services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/ControllerExceptionHandler.kt

```diff
@@ -157,6 +158,14 @@ class ControllerExceptionHandler {
         return HankeError.HAI2003
     }
 
+    @ExceptionHandler(HakemusInWrongStatusException::class)
+    @ResponseStatus(HttpStatus.CONFLICT)
+    @Hidden
+    fun hakemusInWrongStatusException(ex: HakemusInWrongStatusException): HankeError {
+        logger.warn(ex) { ex.message }
+        return HankeError.HAI2015
+    }
+
```

Comment:
Unrelated to this but maybe we could delete the obsolete exception handling of contact processing restrictions (lines 67-79 + `HankeError.HAI1029` enum):
```
    @ExceptionHandler(HankeYhteystietoProcessingRestrictedException::class)
    // Using 451 (since the restriction is typically due to legal reasons).
    // However, in some cases 403 forbidden might be considered correct response, too.
    @ResponseStatus(HttpStatus.UNAVAILABLE_FOR_LEGAL_REASONS)
    @Hidden
    fun hankeYhteystietoProcessingRestricted(
        ex: HankeYhteystietoProcessingRestrictedException
    ): HankeError {
        logger.warn { ex.message }
        // TODO: the response body SHOULD include an explanation and link to server;
        //  left as future exercise. See https://tools.ietf.org/html/rfc7725
        return HankeError.HAI1029
    }
```